### PR TITLE
Fix content length of file parameter

### PR DESCRIPTION
**Pull request by dgreenbean**
Adding required input of content length when adding a file using a writer action.

### CODE REVIEW

**Comment by dgreenbean** [comment]
This is to address issue https://github.com/restsharp/RestSharp/issues/742.

File: RestSharp/IRestRequest.cs [hunk 1]
```diff
@@ -158,9 +158,10 @@ public interface IRestRequest
         /// <param name="name">The parameter name to use in the request</param>
         /// <param name="writer">A function that writes directly to the stream.  Should NOT close the stream.</param>
         /// <param name="fileName">The file name to use for the uploaded file</param>
+	/// <param name="contentLength">The length (in bytes) of the file content.</param>
```
Comment (hallem) [inline RestSharp/IRestRequest.cs]: Please fix the spacing difference here.

File: RestSharp/RestRequest.cs [hunk 2]
```diff
@@ -172,15 +172,17 @@ public IRestRequest AddFile(string name, byte[] bytes, string fileName, string c
         /// <param name="name">The parameter name to use in the request</param>
         /// <param name="writer">A function that writes directly to the stream.  Should NOT close the stream.</param>
         /// <param name="fileName">The file name to use for the uploaded file</param>
+	/// <param name="contentLength">The length (in bytes) of the file content.</param>
```
Comment (hallem) [inline RestSharp/RestRequest.cs]: Please fix the spacing difference here.

**Comment by hallem** [comment]
Code looks good. Just some minor things that need to be changed first.  Also, any idea on why this broke?  I haven't looked into it too deeply yet but I'm concerned about changing the method signature and breaking other implementations.

**Comment by dgreenbean** [comment]
I didn't look too closely at why this broke.  Since the method is currently broken anyway, I wasn't too concerned about changing the signature.  If you think the compiler errors could cause problems for people, I could make this method throw an Exception and add a new method that works.  That felt like overkill to me.

**Comment by dgreenbean** [comment]
The spacing should be fixed now.  I'm guessing at the moment that the AppVeyor failure is unrelated to the change.  Please let me know if this is not the case.

**Comment by hallem** [comment]
it is unrelated unless it doesn't run successfully locally :)

**Comment by dgreenbean** [comment]
@hallem Is there anything blocking this from being merged?

**Comment by hallem** [comment]
no i just haven't had time to sit down and work on this.  i've recently moved so i'm still dealing with all that.  i should have some time soon.

**Pull request merged**